test/test_ops.py

```diff
@@ -1355,6 +1366,56 @@ def test_split_normalization_params(self, norm_layer):
 
         assert len(params[0]) == 92
         assert len(params[1]) == 82
+        
+        
+class TestDropBlock:
+    @pytest.mark.parametrize("seed", range(10))
+    @pytest.mark.parametrize("dim", (2, 3))
+    @pytest.mark.parametrize("p", (0, 0.2, 0.5, 0.8))
+    @pytest.mark.parametrize("block_size", [5, 7, 9, 11])
+    @pytest.mark.parametrize("inplace", [True, False])
+    def test_drop_block(self, seed, dim, p, block_size, inplace):
```

Comment:
The testing strategy here is not great as we only validate the input for a fraction of the inputs. An alternative approach would be to try to count the zeroes based on the sampling probability. As noted on the paper, the number of zeroes will be approximate due to overlaps but perhaps we can create a not so flaky test for this.

If you prefer, we can do this on a follow up PR.

Reply:
I have read the test function of  https://github.com/pytorch/vision/pull/4301, is it necessary check this? I found there is no such check for `nn.Dropout` in PyTorch.

Reply:
Yes that's what I had in mind. The test of `StochasticDepth` confirms that the number of drops are within the expended confidence intervals. These intervals should be larger at DropBlock because of the overlapping blocks.

Concerning whether this test is necessary, the answer is I wouldn't block the PR for this but it will protect us from someone breaking the op accidentally on the future. I'm surprised you say that `Dropout` is not properly checked at PyTorch Core. Perhaps the test is not on an obvious location (I know dropout has tests in a few different places)?

Reply:
@datumbox oh, maybe I missed some codes.

Reply:
@datumbox I add the random test check. But, just as you said, the actual sampling probability is slightly lower than the setting value because of overlaps in dropblock.